.github/actions/restore-node/action.yml

```
@@ -21,6 +21,40 @@ runs:
         clean: 'false'
         submodules: 'true'
         persist-credentials: false
+    # Select a branch on Endo to test against by adding text to the body of the
+    # pull request. For example: #endo-branch: some-pr-branch
+    # The default is '*NONE*' to indicate not to check out Endo, just use
+    # the published NPM packages.
+    - name: Get the appropriate Endo branch
+      id: endo-branch
+      uses: actions/github-script@v6
+      with:
+        result-encoding: string
+        script: |-
+          let branch = 'NOPE';
+          if (context.payload.pull_request) {
+            const { body } = context.payload.pull_request;
+            const regex = /.*\#endo-branch:\s+(\S+)/;
+            const result = regex.exec(body);
+            if (result) {
+              branch = result[1];
+            }
+          }
+          console.log(branch);
+          return branch;
+    - name: check out Endo if necessary
+      uses: actions/checkout@v3
+      with:
+        repository: agoric/endo
+        path: ./replacement-endo
+        ref: ${{ steps.endo-branch.outputs.result }}
+        clean: 'false'
+        submodules: 'true'
+        persist-credentials: false
+      if: steps.endo-branch.outputs.result != 'NOPE'
+    - name: Move Endo checkout outside the working directory
+      run: |-
+        test ! -e ./replacement-endo || mv ./replacement-endo ~/endo
```

What's the reason to use `~` instead of a relative path? Wouldn't we end up with nested repos again if the cwd is actually `~`?
```suggestion
        test ! -e ./replacement-endo || mv ./replacement-endo ../endo
```

---

In Github Actions, the default cwd is not `~`, it is `~/agoric-sdk`.  When we check out to relative subdirs, we get paths like `~/agoric-sdk/agoric-sdk` and `~/agoric-sdk/dapp`.  I don't want to make this configuration relative to the subpath, because then it is still kept within `~/agoric-sdk`.